padd.sh

```diff
@@ -104,6 +104,9 @@ ConstructAPI() {
 	# If no arguments were supplied set them to default
 	if [ -z "${URL}" ]; then
 		URL=127.0.0.1
+        # when no $URL is set we assume PADD is running locally and we can get the port value from FTL directly
+        PORT="$(pihole-FTL  --config webserver.port)"
+        PORT="${PORT%%,*}"
 	fi
 	if [ -z "${PORT}" ]; then
 		PORT=80
```

Comment:
Why fall back to PORT=80 at all? If pihole-FTL cannot tell you it's own port from the command above, then we've got bigger problems. Seems a waste of time then setting it to 80?

Reply:
This is the fallback if `$URL` is not supplied but no `port`. E.g. Remote access it, but use default port. We could also make this an `else`to `URL`

Reply:
Ah, yes, fair point - I keep forgetting http APIs can be accessed from other machines. Carry on!

Reply:
.. thinking about it, it can't be `else` to `URL`...